src/app/cards/components/search-page/search-page.component.ts

```diff
@@ -1,31 +1,68 @@
-import { Component } from '@angular/core';
-import { FormBuilder, FormGroup } from '@angular/forms';
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { FormBuilder, FormControl, FormGroup } from '@angular/forms';
+import { Observable, Subject } from 'rxjs';
+import { map, startWith, switchMap, takeUntil, tap } from 'rxjs/operators';
 import { Store } from '@ngrx/store';
 
 import * as fromRoot from '@/app/store';
 
+import { SetsService } from '@/app/core/services/sets.service';
+
 @Component({
   selector: 'tmm-search-page',
   templateUrl: './search-page.component.html',
   styleUrls: ['./search-page.component.scss']
 })
-export class SearchPageComponent {
+export class SearchPageComponent implements OnInit, OnDestroy {
   public searchForm: FormGroup;
+  public filteredSets: Observable<any[]>;
+  public setSearch = new FormControl('');
+
+  private sets: any[] = [];
+  private destroy = new Subject<void>();
 
   constructor(
     private store: Store<fromRoot.State>,
+    private setsService: SetsService,
     formBuilder: FormBuilder
   ) {
     this.searchForm = formBuilder.group({
       search: '',
       standard_legal: false,
+      modern_legal: false,
+      set: ''
     });
   }
 
+  public ngOnInit(): void {
+    this.filteredSets = this.setsService.index().pipe(
+      switchMap((resp) => {
+        this.sets = resp.sets;
+        return this.setSearch.valueChanges
```

Comment:
Get the sets first, then switch to the filtering form control to control what the dropdown actually displays.